src/nvim/window.c

```diff
@@ -2694,17 +2694,18 @@ void win_free_all(void)
   while (first_tabpage->tp_next != NULL)
     tabpage_close(TRUE);
 
+  if (aucmd_win != NULL) {
+    win_remove(aucmd_win, NULL);
```

Comment:
I don't think `win_remove` is necessarily idempotent.  Could check if `wp->w_prev != NULL` before calling `win_remove (should be enough as aucmd_win cannot be first).

Reply:
Doesn't `win_remove(aucmd_win, NULL)` guarantee that `aucmd_win` is not encountered in the `while (lastwin != NULL && lastwin->w_floating)` loop? It introduced more `heap-use-after-free` bugs though, the ASAN build fails on Travis in the `win_remove(aucmd_win, NULL)` call when `aucmd_win` is manually closed before. Am I missing something?

Reply:
No, I just mean the aucmd_win might already (actually typically) be removed before _this_ line. You have to check if it is in the list before calling `win_remove`.

Reply:
Oh, yes, that's why it fails in some tests